src/components/ShowList.tsx

```diff
@@ -0,0 +1,65 @@
+import IShow from "../utils/i-show";
+import { omitTags } from "../utils/omit-tags";
+import { useEffect, useState } from "react";
+
+interface ShowListProps {
+  setSelectedShow: React.Dispatch<React.SetStateAction<number | null>>;
+  show: IShow;
+}
+
+export default function ShowList({
+  setSelectedShow,
+  show,
+}: ShowListProps): JSX.Element {
+  const [isFavourite, setIsFavourite] = useState<boolean>(() => {
```

Comment:
I don't know enough about this but I get that you're setting the initial state for the useState as either true or false.
Mostly because I don't know much about working the localStorage so would appreciate some comments here as a noob.

Reply:
Basically, for each show id, we are storing in local storage whether it is true or false that that show id is a favourite.